packages/plugin-ext/src/main/browser/editors-and-documents-main.ts

```diff
@@ -384,7 +394,7 @@ class EditorAndDocumentState {
 
 class EditorSnapshot {
     readonly id: string;
-    constructor(readonly editor: MonacoEditor) {
+    constructor(readonly editor: MonacoEditor | SimpleMonacoEditor) {
```

Comment:
Suggestion: We could simply use the underlying type of `getControl()` (which is `ICodeEditor`). Using the union everywhere seems to lead to a lot of code duplication.

Reply:
that does not work because both of them are inheriting from different ICodeEditor interfaces which both use an enum with the same name but different implementation somewhere deep inside. So the getControl of both of them is not compatible

Reply:
or maybe not. Maybe my IDE is just confused out of some weird reason

Reply:
SimpleMonacoEditor uses the ICodeEditor from editorBrowser.ts and the MonacoEditor uses it from the monac editor api directly

Reply:
Oh, ok I see. Makes sense.